### PR TITLE
feat(i18n): Add Japanese translation

### PR DESCRIPTION
**Pull request by sidemt**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)

- Added Japanese translations to the strings in `locales/ja.json`
  - In `keyword-meta` I included an extra keyword "フリーコードキャンプ" which is "freeCodeCamp" written in Japanese characters.

- Updated `assets/config/ja/main.js`, but contents other than forum are not ready in Japanese yet, sot I left them unchanged.

### CODE REVIEW

File: locales/ja.json [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
-  "footer-shop": "Shop",
-  "footer-support": "Support",
-  "footer-sponsors": "Sponsors",
-  "footer-honesty": "Academic Honesty",
-  "footer-coc": "Code of Conduct",
-  "footer-privacy": "Privacy Policy",
-  "footer-tos": "Terms of Service",
-  "footer-copyright": "Copyright Policy"
+  "keyword-meta": "freeCodeCamp, フリーコードキャンプ, プログラミング, フロントエンド, プログラマー, 記事, 正規表現, Python, JavaScript, AWS, JSON, HTML, CSS, Bootstrap, React, Vue, Webpack",
+  "banner": "コーディングを学ぶ — <span>約 3,000 時間のカリキュラムが無料</span>",
```
Comment (majestic-owl448) [inline locales/ja.json:19]: You can set this to link to what you want, with matching translation. The link is the `banner-url` link in the other file. For most languages we set it to link to the chatroom in the specific language.

**Comment by majestic-owl448** [comment]
I think you are missing some strings, look for example at it.json (they are not included in the English file, but they are translatable too)
https://github.com/freeCodeCamp/news-theme/blob/main/locales/it.json

**Comment by sidemt** [comment]
Thanks @ieahleen ! I will check it and make updates.

File: locales/ja.json (hunk 1 above)
Comment (sidemt) [reply locales/ja.json:19]: Thanks for letting me know this. But I think people will be more interested in free coding curriculums even if it's not in Japanese yet. So I'm thinking to leave it as is.

**Comment by sidemt** [comment]
I updated translations. I tested it locally and found some issues.

- The key `tweet it` in the json file is not rendered properly. I needed to change it to `tweet it.` to make it work. Looks like other languages are having the same issue, so if it is ok I will fix other languages' files too.
- The strings below don't include `.` in translatable string. But in Japanese we end a sentence with `。` instead of `.` 
  We need to include `.` in translatable strings to display it properly in Japanese.
`"Read <a href='{url}'>more posts</a> by this author"`
`"Thank you {twitter} for writing this helpful article"`
スクリーンショット 2021-10-31 002134
スクリーンショット 2021-10-31 000847

**Comment by scissorsneedfoodtoo** [comment]
Thanks for catching these issues for the new and existing locales, @sidemt.

Yes, if you could go back and change the `tweet it` keys to `tweet it.` in `es.json`, `it.json`, and `pt-br.json`, that would be great.

I do like the solution you came up with for the Japanese localization, though, with a full sentence followed by a button without punctuation. I think it would be good to do that for the other locales, but that can be handled later with another PR. Just adding the period to the keys to get it working now should be fine.

About the `"Thank you {twitter} for writing this helpful article"` and `"Read <a href='{url}'>more posts</a> by this author"` sentences, if you could include the period in the keys for those, and update the translated string to include the correct period, that would be a huge help.

I believe both keys / translated strings need to be updated in `es.json`, `it.json`, `ja.json`, and `pt-br.json` need to be update. But in `zh-cn.json`, only the `"Read <a href='{url}'>more posts</a> by this author"` key and its string need to be updated -- we don't show the Twitter buttons for that locale.

Also, in `partials/social-row.hbs`, this line:

```js
const thanks = `{{encode (t "Thank you {twitter} for writing this helpful article" twitter=primary_author.twitter)}}.` +
            `%0A%0A${title}%0A%0A${url}`;
```

Will need to be updated so the period is included in `"Thank you {twitter} for writing this helpful article"`.

Thanks again for all of your hard work and thorough checking on all this. Please let me know if there's anything I can do to help.

**Comment by sidemt** [comment]
@scissorsneedfoodtoo Updated. Could you please check them? Thank you.

**Review by scissorsneedfoodtoo — APPROVED** [review]
Thanks again for your all of your hard work on this @sidemt, and for helping us fix those issues with the other locales.

I double checked all the views and buttons for the new Japanese locale. Also, I checked all the tweet buttons for all the different locales in both states (when an author does and doesn't have a Twitter account in their profile).

Everything LGTM :+1: :+1: 

I'll leave this PR up for another reviewer in case I missed something.

But since the Japanese publication is still in draft mode, I'll go ahead and upload your changes here so you and the other translators can start using the new theme.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**